src/main/scala/lms/thirdparty/mpi.scala

```diff
@@ -0,0 +1,83 @@
+package lms.thirdparty
+
+import lms.core._
+import lms.util._
+import lms.core.stub._
+import lms.core.Backend._
+import lms.core.virtualize
+import lms.core.utils.time
+import lms.macros.SourceContext
+
+import lms.collection._
+trait MPIOps { b: Base with PointerOps =>
+  /* LMS support for MPI library */
+
+  def mpi_init(): Rep[Unit] = Wrap[Unit](Adapter.g.reflectWrite("mpi-init")(Adapter.CTRL))
+
+  abstract class MPIWorld extends Manifest[MPIWorld]
+  lazy val mpi_comm_world: Rep[MPIWorld] = Wrap[MPIWorld](Adapter.g.reflect("INLINE_mpi-comm-world"))
```

Comment:
Why are we using a reflect for a constant?

```
object MPICommWorld {
   override def toString = "MPI_COMM_WORLD"
}
val mpi_comm_world: Rep[MPIWorld] = unit(MPICommWorld)
```

I don't really like the _ in names but I guess that's ok. I would match the exact C name if we wanted to keep the convention, or completely making Scala like. This seems like a middle ground.

Reply:
this is super interesting. Seems a big simplification to using `reflect` and solves the `mustInline` problem above :) 

In order for the type to be correct, I had to make some modifications
  
```
  class MPIWorld extends Manifest[MPIWorld] {
    def runtimeClass = classOf[MPIWorld]
  }

  object MPICommWorld extends MPIWorld {
    override def toString = "MPI_COMM_WORLD"
  }
  lazy val mpi_comm_world: Rep[MPIWorld] = unit(MPICommWorld)
```
does this change make sense?